src/SimplifySpecializations.cpp

```diff
@@ -111,8 +141,110 @@ vector<Definition> propagate_specialization_in_definition(Definition &def) {
 void simplify_specializations(map<string, Function> &env) {
     for (auto &iter : env) {
         Function &func = iter.second;
-        propagate_specialization_in_definition(func.definition());
+        propagate_specialization_in_definition(func.definition(), func.name());
+    }
+}
+
+namespace {
+
+uint16_t vector_store_lanes = 0;
+
+int my_trace(void *user_context, const halide_trace_event_t *ev) {
+    if (ev->event == halide_trace_store) {
+        if (ev->type.lanes > 1) {
+            vector_store_lanes = ev->type.lanes;
+        } else {
+            vector_store_lanes = 0;
+        }
     }
+    return 0;
+}
+
+}  // namespace
+
+void simplify_specializations_test() {
```

Comment:
Even though this uses Internal apis, I think it would be better as a standalone JIT test